Ignore more potentially incorrect data from BASS

peppy opened the pull request:
- Closes https://github.com/ppy/osu/issues/34577.
- See https://www.un4seen.com/forum/?msg=145474.

This is still a workaround but arguably it's something we could leave in place without much loss. I think this at least feels better than the previous code.

Notably, you could argue the test coverage of the fail case is lower since made it implicit that all tests will avoid the "backwards seek" detections.  But we never really had tests correctly- fail on the original so I don't see any loss of value.

Can be tested using

```diff
diff --git a/osu.Game/Screens/Play/GameplayClockContainer.cs b/osu.Game/Screens/Play/GameplayClockContainer.cs
index 2afdcfaebb..430a873e70 100644
--- a/osu.Game/Screens/Play/GameplayClockContainer.cs
+++ b/osu.Game/Screens/Play/GameplayClockContainer.cs
@@ -9,6 +9,7 @@
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Logging;
 using osu.Framework.Timing;
+using osu.Framework.Utils;
 using osu.Game.Beatmaps;
 
 namespace osu.Game.Screens.Play
@@ -188,12 +189,20 @@ double IAdjustableClock.Rate
 
         public double Rate => GameplayClock.Rate;
 
-        public double CurrentTime => GameplayClock.CurrentTime;
+        private bool badDataFrame;
+
+        public double CurrentTime => (badDataFrame ? 10000 : 0) + GameplayClock.CurrentTime;
 
         public bool IsRunning => GameplayClock.IsRunning;
 
         #endregion
 
+        protected override void Update()
+        {
+            base.Update();
+            badDataFrame = RNG.NextDouble() > 0.999;
+        }
+
         public void ProcessFrame()
         {
             // Handled via update. Don't process here to safeguard from external usages potentially processing frames additional times.

```

peppy committed:
Ignore more potentially incorrect data from BASS

This is still a workaround but arguably it's something we could leave in
place without much loss. I think this at least feels better than the
previous code.

Notably, you could argue the test coverage of the fail case is lower
since made it implicit that all tests will avoid the "backwards seek"
detections.  But we never really had tests correctly- fail on the original
so I don't see any loss of value.

bdach reviewed (commented):
as discussed on discord the last commits wrt `AllowReferenceClockSeeks` are getting replaced by a (transitionary) debug check instead so I'm not going to comment on those changes

peppy committed:
Don't enforce hotfix during debug builds, to allow interactive tests to run correctly

peppy committed:
Fix half written comment and adjust threshold down slightly

bdach committed:
Remove failing test

Test was added in 5495c2090a76bfd6f6a5efd0fd61530395cc6a68 to exercise a
previous version of the hotfix-workaround of not allowing seeking back.
Now that the hotfix-workaround also encompasses seeking *forward*, which
breaks another 100 tests, thus requiring the hotfix-workaround to be
turned off for tests, this situation is just completely out of
control. So I'm just removing the test because ugh.
